[KLUSH 19기 김현영]Refactor:수정 요청 반영하여 CategoryView에 병합

hyun0k opened the pull request:
>wecode PR message template 및 체크 리스트 입니다.
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- 검색 엔드포인트 구현
		 
## 수정 사항들 자세한 내용
- 중복 코드 방지를 위해 CategoryView안에 검색 기능 병합
- search 키 값이 들어오면 상품이름에 해당 값이 있는 상품 데이터 전달
- 정렬 값 매칭 방식 원래대로 수정

### 테스트 결과
![Screen Shot 2021-04-21 at 19 21 58](https://user-images.githubusercontent.com/38034610/115538531-f27e5a00-a2d6-11eb-9f51-8b2a8f446fcf.png)

## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hyun0k committed:
Add:검색 엔드포인트 추가
- 상품 이름에 keyword가 포함되면 해당 상품 정보를 전달하는 엔드포인트

hyun0k committed:
Add:검색 엔드포인트 추가에 따른 url 추가

redsaumon commented on products/views.py:
```diff
@@ -99,3 +99,25 @@ def get(self, request):
+class SearchView(View):
+    def get(self, request):
+        try:
+            keyword = request.GET.get('keyword')
```
현영님 좋습니다. 그런데 검색, 정렬 등의 기능은 어떤 parameter를 사용하는지 기억하시나요? RESTful API 다시 복습하시고 수정해주세요!

hyun0k replied:
네 알겠습니다!

hyun0k committed:
Refactor:수정 요청 반영하여 parameter명 수정
- 검색 parameter를 search로 변경
- CategoryView의 정렬 parameter를 ordering으로 변경
- 정렬값을 RESTful하게 변경

hyun0k committed:
Merge branch 'master' into feature/search

conflict 해결

redsaumon commented on products/views.py:
```diff
@@ -70,20 +70,13 @@ def get(self, request):
 
-            if(sort_type is not None):
-                product_list = product_list.order_by(sort_list[sort_type])
+            if sort_type:
+                product_list = product_list.order_by(sort_type)
```
들어오는 키워드명과 컬럼명이 일치할 경우 이렇게 편하게 넣을 수 있지만, 테이블 정보가 노출되어있다는 의미이기도 합니다.

hyun0k replied:
- 한 개의 View에 너무 많은 기능이 섞여있으면 좋지 않다고 생각했습니다.
- 기능적으로도 CategoryView는 상품페이지에서 주로 동작하고 검색은 메인페이지의 검색바에서 동작하기 때문에 분리하면 좋을 것 같아서 분리했습니다.
- RESTful한 방식이라고 생각해서 로직을 수정했는데 좋지 않은 방법인가요?

redsaumon replied:
상품페이지에서는 검색 기능을 지원하지 않나요?

hyun0k replied:
상품페이지 안에 검색하는 기능은 없고 nav바에서 검색을 하는 방식입니다.

redsaumon reviewed (commented):
현영님 좋습니다~ 그런데 SearchView와 CategoryView에 중복된 코드가 많이 보여요. 필터링 로직을 CategoryView에 합치지 않고 분리하신 이유가 있으실까요?

hyun0k committed:
Refactor:SearchView 삭제에 따른 url삭제

hyun0k committed:
Refactor:SearchView 삭제
- 중복 코드 방지를 위해 CategoryView에 병합
